### PR TITLE
ML model retrain

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- retrain ML model with templates for .tfvars
- use random generation for train data
- use auto-balanced classes in train
- unit tests refactoring due almost 0 probability for some cases and macos variation
- Actualize BM scores
- Actualize split in benchmark and use Stratified Sampling for test set

Train plots:

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -348,7 +351,8 @@ jobs:
       - name: Checkout CredData
         uses: actions/checkout@v3
         with:
-          repository: Samsung/CredData
+          repository: babenek/CredData
+          ref: auxiliary
```
Comment (babenek) [inline .github/workflows/benchmark.yml:355]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -171,7 +173,8 @@ jobs:
       - name: Checkout CredData
         uses: actions/checkout@v4
         with:
-          repository: Samsung/CredData
+          repository: babenek/CredData
+          ref: auxiliary
```
Comment (babenek) [inline .github/workflows/benchmark.yml:177]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -66,7 +67,8 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:71]:
```suggestion
          repository: Samsung/CredData
```

File: .github/workflows/benchmark.yml [hunk 4]
```diff
@@ -22,7 +22,8 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:26]:
```suggestion
          repository: Samsung/CredData
```

**Review by babenek — COMMENTED** [review]
TODO: rollback after https://github.com/Samsung/CredData/pull/111 merge

File: cicd/benchmark.txt [hunk 5]
```diff
@@ -1,15 +1,15 @@
-DATA: 19076375 valid lines. MARKUP: 64556 items
+DATA: 19076375 valid lines. MARKUP: 64560 items
```
Comment (babenek) [inline cicd/benchmark.txt:1]:
```suggestion
DATA: 19076375 valid lines. MARKUP: 64562 items
```

File: cicd/benchmark.txt [hunk 6]
```diff
@@ -228,20 +228,20 @@ FileType           FileNumber    ValidLines    Positives    Negatives    Templat
...
+Authentication Credentials    83     8      2655    13  0.003004  0.135417  0.992389  0.912088  0.864583  0.887701
+Cryptographic Primitives      50     0       174     4            0.074074  0.982456  1.000000  0.925926  0.961538
+Generic Secret              1123    19     30184    87  0.000629  0.071901  0.996626  0.983363  0.928099  0.954932
+Generic Token                312     5      4264    31  0.001171  0.090379  0.992194  0.984227  0.909621  0.945455
+Other                        546   322      3418   184  0.086096  0.252055  0.886801  0.629032  0.747945  0.683354
+Password                    1269   119     11248   220  0.010469  0.147750  0.973631  0.914265  0.852250  0.882169
+Predefined Pattern           413     3      5298    14  0.000566  0.032787  0.997032  0.992788  0.967213  0.979834
 Private Key                 1019     0      1477     0                      1.000000  1.000000  1.000000  1.000000
-                            4677   416  19070614   668  0.000022  0.124977  0.999943  0.918319  0.875023  0.896149
+                            4815   476  19070531   553  0.000025  0.103018  0.999946  0.910036  0.896982  0.903462
```
Comment (babenek) [inline cicd/benchmark.txt:247]:
```suggestion
                            4815   476  19070531   553  0.000025  0.103018  0.999946  0.910036  0.896982  0.903462

```

**Review by Yullia — APPROVED** [review]
Approved

File: tests/data/ml_threshold.json [hunk 7]
```diff
@@ -9192,7 +9092,7 @@
     {
         "api_validation": "NOT_AVAILABLE",
         "ml_validation": "VALIDATED_KEY",
-        "ml_probability": 0.00144,
```
Comment (babenek) [inline tests/data/ml_threshold.json]: before the train

**Review by Yullia — APPROVED** [review]
Great job!

**Review by iuriimet — APPROVED** [review]

**Pull request merged**